Dev modify tbox generator to support new iri schema

msff2 opened the pull request.

markushofmeister reviewed (changes requested):
I tested it with the updated OntoTimeSeries TBox on `dev-add-forecasting-ontology` and it works. However, a few comments:
1) There is currently no documentation on how to use the tool (Or I am unable to find it). Please add, at least some high level description how to use it
2) It only supports certain .csv types, i.e. UTF-8 encoded .csv (as saved by Excel) currently don't work
3) Regarding the supported XSD types: In case I enter potentially not supported types (e.g. `test`), they are just converted to `string` without any warning. That is risky I believe and should at least output a warning or something. In case I would enter `https://www.w3.org/2001/XMLSchema#test`, it would just get accepted. Not sure if that is correct behaviour (at least without any further documentation)

Lastly, would those changes also require any adjustments to the ABox Generator?

jb2197 commented:
> 1. There is currently no documentation on how to use the tool (Or I am unable to find it). Please add, at least some high level description how to use it

My two cents if you don't mind, could you please include in the documentation the codes to generate TBox OWL from python? An example:

```python
from py4jps.resources import JpsBaseLib

jpsBaseLibGW = JpsBaseLib()
jpsBaseLibGW.launchGateway()
jpsBaseLib_view = jpsBaseLibGW.createModuleView()
jpsBaseLibGW.importPackages(jpsBaseLib_view, "uk.ac.cam.cares.jps.base.converter.*")
tbox_generation = jpsBaseLib_view.TBoxGeneration()
# NOTE replace './ontosomething.csv' with the desired ontology
tbox_generation.generateTBox('./ontosomething.csv')
```

This would really help people update ontologies quickly in WSL.

msff2 commented:
> I tested it with the updated OntoTimeSeries TBox on `dev-add-forecasting-ontology` and it works. However, a few comments:
> 
> 1. There is currently no documentation on how to use the tool (Or I am unable to find it). Please add, at least some high level description how to use it
> 2. It only supports certain .csv types, i.e. UTF-8 encoded .csv (as saved by Excel) currently don't work
> 3. Regarding the supported XSD types: In case I enter potentially not supported types (e.g. `test`), they are just converted to `string` without any warning. That is risky I believe and should at least output a warning or something. In case I would enter `https://www.w3.org/2001/XMLSchema#test`, it would just get accepted. Not sure if that is correct behaviour (at least without any further documentation)
> 
> Lastly, would those changes also require any adjustments to the ABox Generator?

1. Provided detailed documentation on how to create a TBox CSV file-based template and how to run TBox Generator to convert a CSV template file into OWL.
2. Added a note to indicate that the tool can convert a specific type of CSV file.
3. Added the following warning in the code for unknown data types: "The following data type is not valid:" + <PROVIDED_DATA_TYPE>
					+ ". Now the converter will proceed with the string data type."

msff2 commented:
> > 1. There is currently no documentation on how to use the tool (Or I am unable to find it). Please add, at least some high level description how to use it
> 
> My two cents if you don't mind, could you please include in the documentation the codes to generate TBox OWL from python? An example:
> 
> ```python
> from py4jps.resources import JpsBaseLib
> 
> jpsBaseLibGW = JpsBaseLib()
> jpsBaseLibGW.launchGateway()
> jpsBaseLib_view = jpsBaseLibGW.createModuleView()
> jpsBaseLibGW.importPackages(jpsBaseLib_view, "uk.ac.cam.cares.jps.base.converter.*")
> tbox_generation = jpsBaseLib_view.TBoxGeneration()
> # NOTE replace './ontosomething.csv' with the desired ontology
> tbox_generation.generateTBox('./ontosomething.csv')
> ```
> 
> This would really help people update ontologies quickly in WSL.

Included the code snippet in the Conversion into OWL Section.

markushofmeister commented on JPS_BASE_LIB/src/main/java/uk/ac/cam/cares/jps/base/converter/README.md:
```diff
@@ -0,0 +1,203 @@
+|Material,Class, , , , , ,A substance that contains at least one site or bulk phase.,http://www.theworldavatar.com/ontology/ontokin/OntoKin.owl,Material|
+|RateCoefficient,Class,http://www.theworldavatar.com/ontology/ontocape/chemical_process_system/CPS_behavior/behavior.owl#ReactionRateCoefficient,EQUIVALENT-TO, , , ,The coefficients used to evaluate the reaction rate expression.,http://www.theworldavatar.com/ontology/ontokin/OntoKin.owl,Rate Coefficient |
+|existsIn,Object Property, , ,SitePhase UNION BulkPhase,Material,only,A relation between a site phase or bulk phase and a material in which they exist.,http://www.theworldavatar.com/ontology/ontokin/OntoKin.owl, exists in|
+|containedIn,Object Property, , ,GasPhase UNION Material,ReactionMechanism,exactly 1,A relation that identifies that a gas phase or material is contained in a reaction mechanism.,http://www.theworldavatar.com/ontology/ontokin/OntoKin.owl,contained in|
```
Is that true by now?
I remember that previously Object Properties required 2 row entries in case a cardinality restricyion should be created. Is that solved now?

msff2 replied:
Yes, it is. Now the code can represent the domain range and cardinality restriction from a single line.

markushofmeister reviewed (commented):
I have not re-tested the TBox Generation, but the mentioned changes sound reasonable to me.

markushofmeister reviewed (approved):
I have not re-tested, but all previous comments seem addressed.
